client.go

```diff
@@ -0,0 +1,73 @@
+package client
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"time"
+)
+
+// HostURL - Default omg.lol URL
+const HostURL string = "https://api.omg.lol"
+
+// Client -
+type Client struct {
+	HostURL    string
+	HTTPClient *http.Client
+	Auth       AuthStruct
+}
+
+// AuthStruct -
+type AuthStruct struct {
+	Email  string `json:"email"`
+	ApiKey string `json:"api_key"`
+}
+
+// NewClient -
+func NewClient(email string, api_key string, host ...string) (*Client, error) {
+	c := Client{
+		HTTPClient: &http.Client{Timeout: 10 * time.Second},
+		// Default omg.lol URL
+		HostURL: HostURL,
+	}
+
+	if len(host) > 0 {
+		c.HostURL = host[0]
+	}
+
+	// If email or api_key not provided, return empty client
+	if email == "" || api_key == "" {
```

Comment:
You can add a `required` struct tag and use a library (like [this one](https://github.com/go-playground/validator) to validate an object has all required fields)